app/views/components/_single-image-upload.html.erb

```diff
@@ -0,0 +1,71 @@
+<%
+  error_items ||= []
+  data_attributes ||= {}
+  image_src ||= nil
+  image_alt ||= nil
+
+  root_classes = %w(app-c-single-image-upload govuk-form-group)
+%>
+
+<%= tag.div class: root_classes, data: data_attributes do %>
+  <%= render "govuk_publishing_components/components/heading", {
+    text: "Image (required)",
+    font_size: "l",
+    margin_bottom: 3
+  } %>
+
+  <% if image_src %>
+    <%= render "govuk_publishing_components/components/heading", {
+      text: "Uploaded image",
+      heading_level: 3,
+      font_size: "m",
+      margin_bottom: 3
+    } %>
+
+    <div class="govuk-grid-row app-c-single-image-upload__uploaded-image">
+      <div class="govuk-grid-column-one-quarter">
+        <img src="<%= image_src %>" alt="" class="app-view-edition-images__image">
+      </div>
+
+      <div class="govuk-grid-column-three-quarters">
+        <p class="govuk-body govuk-!-margin-0"><strong>Alt text: </strong><%= image_alt %></p>
+      </div>
+    </div>
+
+    <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible">
+
+    <%= render "govuk_publishing_components/components/file_upload", {
+      label: {
+        text: "Replace image",
+        heading_size: "m"
+      },
+
+      name: "#{name}[image]",
+      hint: "Images must be 960px by 640px",
+      id: "#{id}_image",
+      value: image_src,
+      error_items: error_items
+    } %>
+  <% else %>
+    <%= render "govuk_publishing_components/components/file_upload", {
+      label: {
+        text: "Upload image"
+      },
+
+      name: "#{name}[image]",
+      hint: "Images must be 960px by 640px",
+      id: "#{id}_image",
+      value: image_src,
+      error_items: error_items
+    } %>
+  <% end %>
+
+  <%= render "govuk_publishing_components/components/input", {
+    label: {
+      text: "Image description (alt text)"
+    },
+    name: "#{name}[image_alt_text]",
+    id: "#{id}_image_alt_text",
+    value: image_alt == "" ? "None" : image_alt # form.object.image_alt_text
```

Comment:
would this be nil since if one is passed in not ""?

If that is the case i think this might be slightly cleaner.

```
 value: image_alt.presence || "None"
```

`.presence` returns either the object if one is present or nil if not

Reply:
Yeah that's true, though we do need to display "None" next to the image when there is no alt text set, so I've move that to that point. 
```
<p class="govuk-body govuk-!-margin-0"><strong>Alt text: </strong><%= image_alt == "" ? "None" : image_alt %></p>
```